lib/cloudregister/registerutils.py

```diff
@@ -1133,82 +1133,42 @@ def clean_registries_conf_podman(private_registry_fqdn):
     if failed:
         return False
 
-    if not registries_conf and not failed:
-        logging.info(
-            'Either %s is empty or it has been backed up' %
-            REGISTRIES_CONF_PATH
-        )
-        return True
-
-    modified = False
+    modified_by_us = False
+    # Drop from unqualified-search-registries
     unqualified_search_reg = registries_conf.get(
         'unqualified-search-registries', []
     )
-    if unqualified_search_reg:
-        if private_registry_fqdn:
-            if private_registry_fqdn in unqualified_search_reg:
-                unqualified_search_reg.pop(
-                    unqualified_search_reg.index(private_registry_fqdn)
-                )
-                modified = True
-        else:
-            new_unq_search_reg = []
-            for fqdn in unqualified_search_reg:
-                if 'registry' in fqdn and 'susecloud.net' in fqdn:
-                    modified = True
-                    continue
-                new_unq_search_reg.append(fqdn)
-
-            if modified:
-                unqualified_search_reg = new_unq_search_reg
-
-        if modified:
-            registries_conf['unqualified-search-registries'] = \
-                unqualified_search_reg
-
-    registries = registries_conf.get('registry', [])
-    if registries:
-        if private_registry_fqdn:
-            private_registry = {
-                'location': private_registry_fqdn,
-                'insecure': False
-            }
-            if private_registry in registries:
-                registries.pop(registries.index(private_registry))
-                modified = True
-        else:
-            registry_index = None
-            for reg_index, registry in enumerate(registries):
-                location = registry.get('location', {})
-                if 'registry' in location and 'susecloud.net' in location:
-                    registry_index = reg_index
-
-            if registry_index is not None:
-                registries.pop(registry_index)
-                modified = True
+    if not private_registry_fqdn:
+        private_registry_fqdn = __matches_susecloud(unqualified_search_reg)
+    if private_registry_fqdn in unqualified_search_reg:
+        unqualified_search_reg.remove(private_registry_fqdn)
+        modified_by_us = True
+    registries_conf['unqualified-search-registries'] = unqualified_search_reg
```

Comment:
ditto here, this should be done if we modified the content, right ?

Reply:
ditto too, see my former comment